subsys/mgmt/mcumgr/grp/fs_mgmt/src/fs_mgmt_hash_checksum_sha256.c

```diff
@@ -13,90 +13,46 @@
 #include <mgmt/mcumgr/grp/fs_mgmt/fs_mgmt_config.h>
 #include <mgmt/mcumgr/grp/fs_mgmt/fs_mgmt_hash_checksum_sha256.h>
 
-#if defined(CONFIG_TINYCRYPT_SHA256)
-#include <tinycrypt/constants.h>
-#include <tinycrypt/sha256.h>
+#define PSA_IMPLEMENTATION IS_ENABLED(CONFIG_MBEDTLS_PSA_CRYPTO_CLIENT)
+
+#if PSA_IMPLEMENTATION
```

Comment:
Why don't use `#ifdef CONFIG_MBEDTLS_PSA_CRYPTO_CLIENT`  directly ? It is easier to see what is the build option

Reply:
I thought that `PSA_IMPLEMENTATION` would be more self-explanatory, also because it is repeated in this source file. But I can change it to only using `CONFIG_MBEDTLS_PSA_CRYPTO_CLIENT`.

Reply:
the ifdef looks cleaner